pyk/src/pyk/proof/reachability.py

```diff
@@ -444,12 +444,7 @@ def as_rules(self, priority: int = 20, direct_rule: bool = False) -> list[KRule]
             or (self.admitted and not self.kcfg.predecessors(self.target))
         ):
             return [self.as_rule(priority=priority)]
-        _rules = []
-        for _edge in self.kcfg.edges():
-            _rule = _edge.to_rule(self.rule_id, priority=priority)
-            assert type(_rule) is KRule
-            _rules.append(_rule)
-        return _rules
+        return [rule for rule in self.kcfg.to_rules(self.rule_id, priority=priority) if isinstance(rule, KRule)]
```

Comment:
Consider

```suggestion
        def assert_rule(x: object) -> KRule:
            assert isinstance(x, KRule)
            return x
            
        return [assert_rule(rule) for rule in self.kcfg.to_rules(self.rule_id, priority=priority)]
```

Reply:
But you can even strengthen the return type of `to_rules` by pulling the assertion there.

Reply:
Although `to_rules` always returns `KRule`, I’m not sure if the previous design was intentional. So I’ve decided to follow your suggestion.